### PR TITLE
enhance karmadactl describe command

### PR DESCRIPTION
**Pull request by hulizhe**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Enhance the `karmadactl describe` command to add the ability to view Karmada control plane resources.
```bash
$ karmadactl describe pods --operation-scope members --cluster member1
Name:             nginx
Namespace:        default
Priority:         0
Service Account:  default
Node:             member1-control-plane/172.18.0.3
... ...
$ karmadactl describe pods 
Name:         nginx
Namespace:    default
Priority:     0
Node:         <none>
Labels:       propagationpolicy.karmada.io/permanent-id=d698e260-0eec-4dd7-b571-9c0cedb158f6
Annotations:  propagationpolicy.karmada.io/name: nginx-847459f8b6
              propagationpolicy.karmada.io/namespace: default
Status:       Running
... ...

$ karmadactl describe pods --operation-scope members                  
error: must specify a member cluster
```

**Which issue(s) this PR fixes**:
Parts of #5249

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
`karmadactl`:  The `describe` command can show details of Karmada resources now.
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
Hi @hulizhe Please rebase this PR since #5254 has been merged and resolve the conflicts.

**Comment by hulizhe** [comment]
> Hi @hulizhe Please rebase this PR since #5254 has been merged and resolve the conflicts.

done~

File: pkg/karmadactl/describe/describe.go [hunk 1]
```diff
@@ -43,21 +43,24 @@ var (
 		prefixed with NAME_PREFIX.`)
 
 	describeExample = templates.Examples(`
+		# Describe a pod in Karmada control plane
+		%[1]s describe pods/nginx
```
Comment (RainbowMango) [inline pkg/karmadactl/describe/describe.go:47]: Can this example go with deployments? As usually no pods in the Karmada control plane?

File: pkg/karmadactl/describe/describe.go [hunk 2]
```diff
@@ -101,23 +109,39 @@ type CommandDescribeOptions struct {
...
+	o.KubectlDescribeOptions, err = describeFlag.ToOptions(parentCommand, args)
 	if err != nil {
 		return err
 	}
-	describeFlag.Factory = memberFactory
-	o.KubectlDescribeOptions, err = describeFlag.ToOptions(parentCommand, args)
+	return nil
+}
+
+// Validate check if the parameters are valid
```
Comment (RainbowMango) [inline pkg/karmadactl/describe/describe.go:136]:
```suggestion
// Validate checks if the parameters are valid
```

**Review by RainbowMango — COMMENTED** [review]
Only some nits, otherwise LGTM.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Comment by RainbowMango** [comment]
/hold 
Please add a release note

**Comment by hulizhe** [comment]
> /hold Please add a release note

done~

**Pull request merged**